include/LLVMSPIRVOpts.h

```diff
@@ -139,13 +145,24 @@ class TranslatorOpts {
 
   FPContractMode getFPContractMode() const { return FPCMode; }
 
+  bool isUnknownIntrinsicAllowed(llvm::IntrinsicInst *II) const noexcept {
```

Comment:
Since this is a non-trivial implementation, I would prefer to move the definition into a .cpp file.  That would also reduce the number of includes needed.

Reply:
OK, I'll create a .cpp file for this purpose, and try forward-declaring as much LLVM types as possible.

Reply:
Done. I was able to get rid of all includes but the `Optional` one (pun intended) due to incomplete-typed field initialization issues.
An additional thought I had was to replace `Optional` with `std::unique_ptr` and use `get()` instead of `hasValue()` and `getValue()`. But that wouldn't seem to gain anything besides the additional smart pointer headache - neither would replacing a relatively small LLVM include with `<memory>` present much advantage.

Reply:
UPD: also forward-declared `Optional` and moved the implementation file to be linked within `libLLVMSPIRV`.

Reply:
Thanks @AGindinson! I think there's not much more to gain indeed, so this looks good to me.